modules/virt-creating-a-service-backed-by-virtual-machines.adoc

```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-creating-a-service-backed-by-virtual-machines_{context}"]
+
+= Creating a service backed by virtual machines
+
+To create a service backed by a virtual machine, you start the `VirtualMachineInstance`,
+if it is not already running. Once the instance is running, you connect to the instance, and
+create a service object to expose the instance.
+
+The following service types are supported:
+
+* `ClusterIP` - Connect using service IP and service port inside the cluster network.
+* `NodePort` - Connect using node IP and node port outside the cluster network.
+* `LoadBalancer` - Connect using public IP and port.
+
+This procedure presents an example of how to create and expose a service object of type `ClusterIP`
+as a virtual machine-backed service.
+
+[NOTE]
+====
+Labels on a `VirtualMachineInstance` are passed through to the pod. The labels on
+the `VirtualMachineInstance` must match the labels on the Service object.
+====
+
+.Procedure
+
+. Edit the `VirtualMachineInstance` YAML configuration file to include the
```

Comment:
If this VMI is already running, then it should be verified whether this step doesn't conflict with the reconciliation initiative (https://issues.redhat.com/browse/CNV-6075).

Reply:
@yossisegev - Can you clarify what needs to be added by 'it should be verified whether this step doesn't conflict with the reconciliation initiative'? Or do you you mean we should tell the user to stop and restart the VMI if it is currently running?

Reply:
@bgaydosrh The second option. the user would have to restart the VMI.
The thing is that when A **VM** is stopped, then there's no **VMI**. Thus, what should be edited is the **VirtualMachine** yaml, and not the **VirtualMachineInstance** yaml.

Reply:
Great, thanks. Sorry... I thought I changed all those VMIs to VMs. WIll fix.

Reply:
Added the following step before the `get service` command:

+
. Start the virtual machine. If the virtual machine is already running, restart it.
+